lib/actors/smart_flare_actor.dart

```diff
@@ -130,6 +133,25 @@ class _SmartFlareActorState extends State<SmartFlareActor> {
     return GestureDetector(
       onTap: () {
         playAnimation(activeArea);
+        if (activeArea.hasAnimationGuard) {
```

Comment:
I can find a better way to do this, I have to get back in the code but I can see what you're doing here. I'd prefer keeping this all in the TapController and not having the SmartFlareActor know about this logic. Time is super tight but it's on my list, just a low priority at the moment.

Reply:
Sure, I know there is a better way, but I needed it for my project sooner, so I implemented a quick-fix-patch.....